### DISCUSSION TITLE
posts/react-query-auth-token-refresh/

### DISCUSSION THREAD

**Discussion by giscus** [yjose/elazizi.com#120] [Blog posts]
# posts/react-query-auth-token-refresh/

A complete guide on how to refresh auth token while using react-query

https://elazizi.com/posts/react-query-auth-token-refresh/

**Comment by SihamBen** [comment]
Great article ! we would appreciate more articles about more complex use cases of tests

**Reply by yjose** [reply]
Mar7ba 😀

**Comment by timmypidashev** [comment]
I genuinely agree with this take. I've speculated on whether to use react-query to perform the actual backend requests for my authentication provider, but there's just no need to add technical debt when a library like axios simplifies the entire useQuery call. It's quite an elegant implementation that avoids unnecessary complexity.

**Comment by tranlehaiquan** [comment]
I think you got cyclic dependencies client.ts and refresh-auth.ts.
